cmake/build_mingw.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+if [ -z "$ARCH" ]; then
+	ARCH=32
+fi
+MINGW=mingw$ARCH
+
+if [ -z "$MSYSCON" ]; then
+	MINGW_PATH=/opt/$MINGW
+
+	DISTRO=$(lsb_release -si)
+
+	if [ "$DISTRO" != "Ubuntu" ]; then
+		echo "This script only supports Ubuntu"
```

Comment:
I know this was here before this PR, but it still may be worth refining... I think this statement is partially wrong. Perhaps we change it to a warning incase another distros compatible with our mingw PPAs are leveraged.

Reply:
Done, I've rewritten the code to just work (also simplifying it a bit on the fly), so now it should (probably) work on some other distros as well. I agree there's no need to fail fast for platforms where it can just work either by design or pure coincidence.